test/test_testing.py

```diff
@@ -778,220 +852,214 @@ def test_quantized_support(self, device):
 
     @onlyCPU
-    def test_assert_tensors_close_only_atol(self, device):
+    def test_assert_close_only_atol(self, device):
         actual = torch.empty((), device=device)
         expected = actual.clone()
 
         with self.assertRaises(UsageError):
-            torch.testing.assert_tensors_close(actual, expected, atol=0.0)
+            torch.testing.assert_close(actual, expected, atol=0.0)
 
     @onlyCPU
-    def test_assert_tensors_close_mismatching_values_rtol(self, device):
+    def test_assert_close_mismatching_values_rtol(self, device):
         eps = 1e-3
         actual = torch.tensor(1.0, device=device)
         expected = torch.tensor(1.0 + eps, device=device)
 
         with self.assertRaises(AssertionError):
-            torch.testing.assert_tensors_close(actual, expected, rtol=eps / 2, atol=0.0)
+            torch.testing.assert_close(actual, expected, rtol=eps / 2, atol=0.0)
 
     @onlyCPU
-    def test_assert_tensors_close_matching_values_rtol(self, device):
+    def test_assert_close_matching_values_rtol(self, device):
         eps = 1e-3
         actual = torch.tensor(1.0, device=device)
         expected = torch.tensor(1.0 + eps, device=device)
 
-        torch.testing.assert_tensors_close(actual, expected, rtol=eps * 2, atol=0.0)
+        torch.testing.assert_close(actual, expected, rtol=eps * 2, atol=0.0)
 
     @onlyCPU
-    def test_assert_tensors_close_mismatching_values_atol(self, device):
+    def test_assert_close_mismatching_values_atol(self, device):
         eps = 1e-3
         actual = torch.tensor(0.0, device=device)
         expected = torch.tensor(eps, device=device)
 
         with self.assertRaises(AssertionError):
-            torch.testing.assert_tensors_close(actual, expected, rtol=0.0, atol=eps / 2)
+            torch.testing.assert_close(actual, expected, rtol=0.0, atol=eps / 2)
 
     @onlyCPU
-    def test_assert_tensors_close_matching_values_atol(self, device):
+    def test_assert_close_matching_values_atol(self, device):
         eps = 1e-3
         actual = torch.tensor(0.0, device=device)
         expected = torch.tensor(eps, device=device)
 
-        torch.testing.assert_tensors_close(actual, expected, rtol=0.0, atol=eps * 2)
+        torch.testing.assert_close(actual, expected, rtol=0.0, atol=eps * 2)
+
+    @onlyCPU
+    def test_mismatching_values_msg(self, device):
+        actual = torch.full((3, 3), 5, dtype=torch.float32, device=device)
+        expected = actual.clone()
+
+        actual[0, 1] = 1
+        expected[0, 1] = 2
+        expected[1, 2] = 9
+
+        expected_regexs = (
+            r"\s+2\s+",  # absolute number of mismatches
+            r"22([.]2+)?\s*[%]",  # relative number of mismatches
+            r"\s+4[.]0\s+",  # maximum absolute difference
+            r"1,\s*2",  # index of maximum absolute difference
+            r"\s+0[.]5\s+",  # maximum relative difference
+            r"0,\s*1",  # index of maximum relative difference
+        )
```

Comment:
After some offline discussion @mattip recommended to not use a regex or worse multiple regexs to check an error message to make the test more readable. I've introduced the regex checking to allow the error message some flexibility in the wording and only check the numbers. Opinion @mruberry?

Reply:
I would also prefer to avoid adding a new regular expression assertion context manager.

One option would be to simplify this test by making it less programmatic and just asserting string equality with an expected message.